src/cli/cli_tcp.hpp

```diff
@@ -0,0 +1,159 @@
+     *
+     * @param[in]  aArgs   An array of command line arguments.
+     *
+     */
+    otError Process(Arg aArgs[]);
+
+private:
+    struct Command
+    {
+        const char *mName;
+        otError (TcpExample::*mHandler)(Arg aArgs[]);
+    };
+
+    otError ProcessHelp(Arg aArgs[]);
+    otError ProcessInit(Arg aArgs[]);
+    otError ProcessDeinit(Arg aArgs[]);
+    otError ProcessBind(Arg aArgs[]);
+    otError ProcessConnect(Arg aArgs[]);
+    otError ProcessSend(Arg aArgs[]);
+    otError ProcessBenchmark(Arg aArgs[]);
+    otError ProcessSendEnd(Arg aArgs[]);
+    otError ProcessAbort(Arg aArgs[]);
+    otError ProcessListen(Arg aArgs[]);
+    otError ProcessStopListening(Arg aArgs[]);
+
+    static void HandleTcpEstablishedCallback(otTcpEndpoint *aEndpoint);
+    static void HandleTcpSendDoneCallback(otTcpEndpoint *aEndpoint, otLinkedBuffer *aData);
+    static void HandleTcpReceiveAvailableCallback(otTcpEndpoint *aEndpoint,
+                                                  size_t         aBytesAvailable,
+                                                  bool           aEndOfStream,
+                                                  size_t         aBytesRemaining);
+    static void HandleTcpDisconnectedCallback(otTcpEndpoint *aEndpoint, otTcpDisconnectedReason aReason);
+    static otTcpIncomingConnectionAction HandleTcpAcceptReadyCallback(otTcpListener *   aListener,
+                                                                      const otSockAddr *aPeer,
+                                                                      otTcpEndpoint **  aAcceptInto);
+    static void                          HandleTcpAcceptDoneCallback(otTcpListener *   aListener,
+                                                                     otTcpEndpoint *   aEndpoint,
+                                                                     const otSockAddr *aPeer);
+
+    void                          HandleTcpEstablished(otTcpEndpoint *aEndpoint);
+    void                          HandleTcpSendDone(otTcpEndpoint *aEndpoint, otLinkedBuffer *aData);
+    void                          HandleTcpReceiveAvailable(otTcpEndpoint *aEndpoint,
+                                                            size_t         aBytesAvailable,
+                                                            bool           aEndOfStream,
+                                                            size_t         aBytesRemaining);
+    void                          HandleTcpDisconnected(otTcpEndpoint *aEndpoint, otTcpDisconnectedReason aReason);
+    otTcpIncomingConnectionAction HandleTcpAcceptReady(otTcpListener *   aListener,
+                                                       const otSockAddr *aPeer,
+                                                       otTcpEndpoint **  aAcceptInto);
+    void HandleTcpAcceptDone(otTcpListener *aListener, otTcpEndpoint *aEndpoint, const otSockAddr *aPeer);
+
+    static constexpr Command sCommands[] = {
```

Comment:
Can we add a `README_TCP.md` that documents these commands and perhaps a quick start section? See [src/cli/README_UDP.md](https://github.com/openthread/openthread/blob/main/src/cli/README_UDP.md) for reference.

Reply:
Sounds good, I will work on this. I'll try to write this up tomorrow. Do you want to include it in this pull request or should I make it a separate pull request?

Reply:
I don't have a preference whether to include in this PR or a separate PR. Thanks.

Reply:
OK, I'll work on it. If, by the time I finish, this pull request is merged, I'll create a new pull request. Otherwise, I'll add it to this one.